junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/config/JupiterConfiguration.java

```diff
@@ -45,4 +47,6 @@
 
 	Predicate<ExecutionCondition> getExecutionConditionFilter();
 
+	Optional<Class<? extends DisplayNameGenerator>> getDefaultDisplayNameGeneratorClass();
```

Comment:
❓ I'm wondering whether we should only instantiate the `DisplayNameGenerator` once and return a `DisplayNameGenerator` instance (with a fallback to `Standard`) from this method instead.

@junit-team/junit-lambda WDYT?

Reply:
Yes, we should only instantiate the default implementation once.

TBH, I haven't looked at the proposal in detail, but I'm wondering if this logic even needs to go into `JupiterConfiguration`. At a glance, it seems potentially misplaced to me.

Reply:
We actually already do that: it's stored in `org.junit.jupiter.engine.descriptor.DisplayNameUtils.standardGenerator`.

So, if we don't find a `@DisplayName`, and we don't find a `@DisplayNameGeneration`, and there's no value for the new configuration parameter, we then just use the static `standardGenerator` field as before.

Reply:
Ahhh... sorry: I misunderstood what you were asking.

Yes, I think we should instantiate the custom default generator only once.

Reply:
@gopinath-langote Could you please change that accordingly?

Reply:
@marcphilipp 
I agree that we can return standard generator from this method if no configuration found or error in getting generator class.

I will make changes